src/HonzaBotner.Discord.Services/EventHandlers/ServerBoosterHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using DSharpPlus.Entities;
+using DSharpPlus.EventArgs;
+using HonzaBotner.Discord.EventHandler;
+using HonzaBotner.Discord.Services.Options;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace HonzaBotner.Discord.Services.EventHandlers
+{
+    public class ServerBoosterHandler : IEventHandler<GuildUpdateEventArgs>
+    {
+        private readonly ILogger<ServerBoosterHandler> _logger;
+        private readonly CommonCommandOptions _commonOptions;
+
+        public ServerBoosterHandler(ILogger<ServerBoosterHandler> logger, IOptions<CommonCommandOptions> commonOptions)
+        {
+            _logger = logger;
+            _commonOptions = commonOptions.Value;
+        }
+
+        public async Task<EventHandlerResult> Handle(GuildUpdateEventArgs eventArgs)
+        {
+            if ((eventArgs.GuildBefore.PremiumSubscriptionCount ?? 0) <=
+                (eventArgs.GuildAfter.PremiumSubscriptionCount ?? 0))
+                return EventHandlerResult.Continue;
+
+            HashSet<ulong> subscribersBefore = new();
+            HashSet<ulong> subscribersAfter = new();
+
+            foreach ((ulong id, DiscordMember member) in eventArgs.GuildBefore.Members)
+            {
+                if (member.PremiumSince.HasValue)
+                {
+                    subscribersBefore.Add(id);
+                }
+            }
+
+            foreach ((ulong id, DiscordMember member) in eventArgs.GuildAfter.Members)
+            {
+                if (member.PremiumSince.HasValue)
+                {
+                    subscribersAfter.Add(id);
+                }
+            }
+
+            HashSet<ulong> newSubscribers = subscribersAfter.Except(subscribersBefore).ToHashSet();
+
+            if (!newSubscribers.Any())
+            {
+                _logger.LogError("No new subscriber even while count after is greater than count before");
+                return EventHandlerResult.Continue;
+            }
```

Comment:
Do we need this, cant we just use the check by counting them?

Reply:
The subscription count is lower in this part, but this check only checks we successfully collected all the new newly subscribed members (probably always 1).

But yes, we can commit this. It should work.